src/sentry/scripts/tsdb/cmsketch.lua

```diff
@@ -318,6 +318,166 @@ function Sketch:increment(items)
     return results
 end
 
+local function response_to_table(response)
+    local result = {}
+    for i = 1, #response, 2 do
+        result[response[i]] = response[i + 1]
+    end
+    return result
+end
+
+function Sketch:export()
+    -- If there's no data, there's nothing meaningful to export.
+    if not self:exists() then
+        return cmsgpack.pack(nil)
+    end
+    return cmsgpack.pack({
+        response_to_table(redis.call('ZRANGE', self.index, 0, -1, 'WITHSCORES')),
+        response_to_table(redis.call('HGETALL', self.estimates)),
+    })
+end
+
+function table.is_empty(t)
+    return next(t) == nil
+end
+
+function Sketch:import(payload)
+    local data = cmsgpack.unpack(payload)
+    if data == nil then
+        return  -- nothing to do here
+    end
+
+    local source_index, source_estimators = unpack(data)
+
+    -- TODO: This is absolutely not efficient when we're merging multiple things into the same sketch.
+    local destination_index = response_to_table(redis.call('ZRANGE', self.index, 0, -1, 'WITHSCORES'))
+    local destination_estimators = response_to_table(redis.call('HGETALL', self.estimates))
+
+    if table.is_empty(source_estimators) and table.is_empty(destination_estimators) then
+        -- If we're just writing the index values (and not estimators) to the
+        -- destination, we can just directly increment the sketch which will
+        -- take care of estimator updates and index truncation.
+        local items = {}
+        for key, value in pairs(source_index) do
+            table.insert(
+                items,
+                {key, tonumber(value)}
+            )
+        end
+        self:increment(items)
+    elseif not table.is_empty(source_estimators) and not table.is_empty(destination_estimators) then
+        -- Merge estimators.
+        for key, value in pairs(source_estimators) do
+            redis.call('HINCRBY', self.estimates, key, value)
+        end
+
+        -- Rebuild the index by using the keys from both indices and the new estimates.
+        local members = {}
+
+        for key, _ in pairs(source_index) do
+            members[key] = true
+        end
+
+        for key, _ in pairs(destination_index) do
+            members[key] = true
+        end
+
+        redis.call('DEL', self.index)
+        for key, _ in pairs(members) do
+            redis.call(
+                'ZADD',
+                self.index,
+                reduce(
+                    math.min,
+                    map(
+                        function (coordinates)
```

Comment:
Yeah, I don't love this but I also don't want to change the rest of the script outside of the scope of this patch just to improve names for now. Willing to come back and look at it later though.